File: Backend/src/main/java/com/example/e_learning_system/Service/Interfaces/MyCoursesService.java
package com.example.e_learning_system.Service.Interfaces;

import com.example.e_learning_system.Dto.MyCourseDtos.MyCoursesResponseDto;
import com.example.e_learning_system.Dto.MyCourseDtos.TeacherCourseDto;
import com.example.e_learning_system.Dto.MyCourseDtos.TeacherStatsDto;

import java.util.List;

/**
 * Service interface for teacher's course management dashboard (My Courses)
 */
public interface MyCoursesService {
    
    /**
     * Get complete dashboard for a teacher
     * @param teacherId The teacher's user ID
     * @return Dashboard with stats and courses
     */
    MyCoursesResponseDto getMyCoursesDashboard(Integer teacherId);
    
    /**
     * Get all courses created by a teacher
     * @param teacherId The teacher's user ID
     * @return List of courses with statistics
     */
    List<TeacherCourseDto> getTeacherCourses(Integer teacherId);
    
    /**
     * Get teacher statistics
     * @param teacherId The teacher's user ID
     * @return Statistics DTO
     */
    TeacherStatsDto getTeacherStats(Integer teacherId);
    
    /**
     * Get recently updated courses
     * @param teacherId The teacher's user ID
     * @param limit Maximum number of courses to return
     * @return List of recently updated courses
     */
    List<TeacherCourseDto> getRecentlyUpdatedCourses(Integer teacherId, int limit);
    
    /**
     * Get top performing courses by enrollment
     * @param teacherId The teacher's user ID
     * @param limit Maximum number of courses to return
     * @return List of top performing courses
     */
    List<TeacherCourseDto> getTopPerformingCourses(Integer teacherId, int limit);
    
    /**
     * Get course details with statistics
     * @param teacherId The teacher's user ID
     * @param courseId The course ID
     * @return Course details with statistics
     */
    TeacherCourseDto getCourseDetails(Integer teacherId, Integer courseId);
}

File: Frontend/src/app/Services/MyCourses/my-courses.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { environment } from '../../../environments/environment';
import { ApiResponse } from '../../Dtos/ApiResponse';
import {
  MyCoursesResponseDto,
  TeacherCourseDto,
  TeacherStatsDto
} from '../../Dtos/MyCourseDtos';

/**
 * Service for teacher's My Courses functionality
 */
@Injectable({
  providedIn: 'root'
})
export class MyCoursesService {
  private api = `${environment.apiUrl}`;

  constructor(private http: HttpClient) {}

  /**
   * Get complete My Courses dashboard for a teacher
   */
  getMyCoursesDashboard(): Observable<MyCoursesResponseDto> {
    return this.http.get<ApiResponse<MyCoursesResponseDto>>(
      `${this.api}/api/my-courses/dashboard`
    ).pipe(
      map(response => response.data)
    );
  }

  /**
   * Get all courses created by a teacher
   */
  getTeacherCourses(): Observable<TeacherCourseDto[]> {
    return this.http.get<ApiResponse<TeacherCourseDto[]>>(
      `${this.api}/api/my-courses/courses`
    ).pipe(
      map(response => response.data)
    );
  }

  /**
   * Get teacher statistics
   */
  getTeacherStats(): Observable<TeacherStatsDto> {
    return this.http.get<ApiResponse<TeacherStatsDto>>(
      `${this.api}/api/my-courses/stats`
    ).pipe(
      map(response => response.data)
    );
  }

  /**
   * Get recently updated courses
   */
  getRecentlyUpdatedCourses(limit: number = 5): Observable<TeacherCourseDto[]> {
    return this.http.get<ApiResponse<TeacherCourseDto[]>>(
      `${this.api}/api/my-courses/recent?limit=${limit}`
    ).pipe(
      map(response => response.data)
    );
  }

  /**
   * Get top performing courses
   */
  getTopPerformingCourses(limit: number = 5): Observable<TeacherCourseDto[]> {
    return this.http.get<ApiResponse<TeacherCourseDto[]>>(
      `${this.api}/api/my-courses/top?limit=${limit}`
    ).pipe(
      map(response => response.data)
    );
  }

  /**
   * Get course details with statistics
   */
  getCourseDetails(courseId: number): Observable<TeacherCourseDto> {
    return this.http.get<ApiResponse<TeacherCourseDto>>(
      `${this.api}/api/my-courses/course/${courseId}`
    ).pipe(
      map(response => response.data)
    );
  }
}
